src/week4/StringAPI.java

```diff
@@ -0,0 +1,142 @@
+package week4;
+
+import static week4.Constants.INVALID;
+import static week4.Constants.OUT_OF_BOUND;
+import static week4.Constants.VALID;
+
+/**
+ * Created by Lilia Shamirian on  28, Nov, 2020
+ */
+public class StringAPI {
+    //Ex1
+    public String getIndexCharacter(String input, int k) {
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUND;
+        }
+        if (k < 0 || k > input.length()) {
+            return OUT_OF_BOUND;
+        }
+        return String.valueOf(input.charAt(k));
+    }
+
+    //Ex2
+    public int getWordCount(String input) {
+        int count = 0;
+        if (input == null || input.isEmpty()) {
+            System.out.println(INVALID);
+            return 0;
+        }
+        if (!(input.matches("(.*) (.*)"))) {
+            System.out.println(" 1 Word ");
+            return 0;
+        }
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == ' ') {
+                count++;
+            }
+        }
+        return count + 1;
+    }
+
+    //Ex3
+    public String checkIfInputIsPalindrome(String input) {
+        StringBuilder st = new StringBuilder(input);
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUND;
+        }
+        if (st.reverse().toString().equals(input)) {
+            return "Palindrome";
+        } else {
+            return "Not Palindrome";
+        }
+    }
+
+    //Ex4
+    public String checkBrackets(String input) {
+        if (input == null || input.isEmpty()) {
+            return INVALID;
+        }
+        int length = input.length();
+        if (length % 2 != 0) {
+            return INVALID;
+        }
+        if (input.matches("[()]+")) {
```

Comment:
also you can do the following to covery other unexpected cases:

if(input.startsWith(")") || input.endsWith("(")) {
      return INVALID;
}

Reply:
ok 👍